include/crm/common/internal.h

```diff
@@ -287,7 +287,8 @@ pcmk__assert_alloc_as(const char *file, const char *function, uint32_t line,
     void *ptr = calloc(nmemb, size);
 
     if (ptr == NULL) {
-        crm_abort(file, function, line, "Out of memory", FALSE, FALSE);
+        crm_abort(file, function, line, "Out of memory", FALSE, TRUE);
+        crm_exit(CRM_EX_OSERR);
```

Comment:
I chose `CRM_EX_OSERR` because that's what `pcmk_rc2exitc()` maps `ENOMEM` to. I don't have a preference.

Reply:
CRM_EX_FATAL causes pacemakerd to shut everything down instead of respawning daemons and causes systemd to not restart pacemakerd when it exits.

We do have a MAX_RESPAWN (100), but a quick look at the code suggests it merely doesn't respawn that child (rather than shut down all of Pacemaker, which would make more sense).

I believe the current code would SIGABRT and respawn, so let's stick with OSERR for now. If we see it happen in the wild we can figure out if something else would be better.

Reply:
It sounds like `CRM_EX_FATAL` does make more sense; I chose `OSERR` for consistency. I'm okay with postponing a change until it becomes an issue, since this should be exceedingly rare and some other crap might continue hitting the fan on an affected system even if Pacemaker shuts down.